### PR TITLE
Fix ShootNotFailed extension predicate

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
The `ShootNotFailed` predicate was flawed and did not work as expected. This PR fixes this and adds proper unit tests to prevent future regressions. Detected as part of the review for #2217, thanks @ialidzhikov for spotting!

**Which issue(s) this PR fixes**:
Related to #2145 ❓

**Special notes for your reviewer**:
/cc @danielfoehrKn, probably is related to #2145 which is what you are currently looking at.

**Release note**:

```improvement developer
The `ShootNotFailed` predicate in the extensions library does now work as expected.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
/kind/bug

File: extensions/pkg/predicate/predicate_test.go [hunk 1]
```diff
@@ -12,25 +12,47 @@
...
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	"github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	mockcache "github.com/gardener/gardener/pkg/mock/controller-runtime/cache"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/golang/mock/gomock"
 	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
```
Comment (timuthy) [inline extensions/pkg/predicate/predicate_test.go:27]: Was that change intended? If yes, why?

**Review by timuthy — COMMENTED** [review]
Looks good, only one question.

File: extensions/pkg/predicate/predicate_test.go (hunk 1 above)
Comment (rfranzke) [reply extensions/pkg/predicate/predicate_test.go:27]: Yes, intended. I changed the package of the `predicate_test.go` file from `predicate_test` to `predicate` to be able to test the unexported function.
Keeping the dot import for `gomega` would now introduce a name conflict as the `predicate` package already defines an `Or` function (similarly to the `gomega` package).

**Review by timuthy — APPROVED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm
